AlarmPlus/Models/Alarm.swift

```diff
@@ -8,183 +8,35 @@
 
 import UIKit
 
-class Alarms: NSObject {
-    private var alarmTime: [Int] = []
-    private var timesSnoozed: Int = 0
-    private var repeatAlarm: Bool = false
-    private var daysActive: [Int] = []
-    private var alarmEnabled: Bool = true
-    private var activeAlarms: [Int : Date] = [ : ]
-    
-    var snooze: Snooze?
-    var alert: Alert?
-    
-    
-    
-    func enableRepeat() {
-        self.repeatAlarm = true
-    }
-    
-    func disableRepeat() {
-        self.repeatAlarm = false
-    }
-    
-    func doesAlarmRepeat() -> Bool {
-        return repeatAlarm
-    }
-    
-    func enableAlarm() {
-        self.alarmEnabled = true
-    }
-    
-    func disableAlarm() {
-        self.alarmEnabled = false
-    }
-    
-    func isAlarmEnabled() -> Bool {
-        return alarmEnabled
-    }
-    
-    func resetSnoozeCount() {
-        self.timesSnoozed = 0
-    }
-    
-    func increaseSnoozeCount() {
-        self.timesSnoozed += 1
-    }
-    
-    func getSnoozeCount() -> Int {
-        return self.timesSnoozed
-    }
-    
-    func addDayToAlarm(daysArray: [Int]) {
-        self.daysActive = daysArray
-    }
-    
-    func getAlarmDays() -> [Int] {
-        return self.daysActive
-    }
-    
-    //    func removeDayFromAlarm(_ day: Int){
-    //        for days in daysActive {
-    //            if day == days {
-    //                daysActive.removeAll{
-    //                    $0 == days
-    //                }
-    //            }
-    //        }
-    //    }
-    
-    func setAlarmTime(hour : Int, minute: Int) {
-        self.alarmTime.append(hour)
-        self.alarmTime.append(minute)
-    }
-    
-    func getAlarms() -> [Int : Date] {
-        return activeAlarms
-    }
+class Alarm: Codable {
+
+
+    var snooze = Snooze()
+    var alert = Alert()
+    var schedule = Schedule()
     
-    func setActiveAlarams() {
-        //adds all of the different scheduled alarms to the activeAlarms array
-        for day in daysActive {
-            addDateToActiveAlarms(day)
-        }
+    init(snooze: Snooze, alert: Alert, schedule: Schedule) {
```

Comment:
to initialize the class with default values, you can write:

`init(snooze: Snooze = Snooze(), alert: Alert = Alert(), schedule: Schedule = Schedule()) {`

What this code says is "if I didn't specify a value for the parameter, use these default values instead.
Then, when you want to create a new Alarm class, you can instantiate it by declaring `Alarm()` it will use the defaults for all values.